src/main/java/picocli/CommandLine.java

```diff
@@ -8030,6 +8053,25 @@ public Help(CommandSpec commandSpec, ColorScheme colorScheme) {
          * may need to re-initialize this field by calling {@link #createDefaultParamLabelRenderer()} again. */
         public IParamLabelRenderer parameterLabelRenderer() {return parameterLabelRenderer;}
 
+        List<IHelpSectionRenderer> sections() {
+        	return Arrays.asList(
+    	        () -> headerHeading(),
+    	        () -> header(),
+    	        () -> synopsisHeading(),      			 //e.g. Usage:
+    	        () -> synopsis(synopsisHeadingLength()), //e.g. &lt;main class&gt; [OPTIONS] &lt;command&gt; [COMMAND-OPTIONS] [ARGUMENTS]
+    	        () -> descriptionHeading(),   			 //e.g. %nDescription:%n%n
+    	        () -> description(),          			 //e.g. {"Converts foos to bars.", "Use options to control conversion mode."}
+    	        () -> parameterListHeading(), 			 //e.g. %nPositional parameters:%n%n
+    	        () -> parameterList(),        			 //e.g. [FILE...] the files to convert
+    	        () -> optionListHeading(),    			 //e.g. %nOptions:%n%n
+    	        () -> optionList(),           			 //e.g. -h, --help   displays this help and exits
+    	        () -> commandListHeading(),   			 //e.g. %nCommands:%n%n
+    	        () -> commandList(),          			 //e.g.    add       adds the frup to the frooble
+    	        () -> footerHeading(),
+    	        () -> footer()
+    	    );
+        }
+        
```

Comment:
We want user to be able to plug in their own section renderers. My initial thought was that users would subclass `Help` and override this method, but now I'm thinking it may be nicer to do something like this:

```
private Map<String, ISectionRenderer> sectionMap = createDefaultSectionMap();

/**
 * Returns the map of section keys and renderers used to construct the usage help message.
 * The usage help message can be customized by adding, replacing and removing section renderers from this map.
 * Sections can be reordered with {@link #setSectionKeys(List) setSectionKeys}.
 * Sections that are either not in this map or not in the list returned by {@link #getSectionKeys() getSectionKeys} are omitted.
 * @since 3.9
 */
public Map<String, ISectionRenderer> getSectionMap() { return sectionMap; }

private Map<String, ISectionRenderer> createDefaultSectionMap() { 
    Map<String, ISectionRenderer> result = new HashMap<String, ISectionRenderer>();

    // TODO we should probably have constants for the default section keys
    result.put("headerHeading", new ISectionRenderer() { public String render(Help help) {
            return help.headerHeading(); } });
    result.put("header", new ISectionRenderer() { public String render(Help help) {
            return help.header(); } });
    result.put("synopsisHeading", new ISectionRenderer() { public String render(Help help) {
            return help.synopsisHeading(); } });
    result.put("synopsis", new ISectionRenderer() { public String render(Help help) {
            return help.synopsis(help.synopsisHeadingLength()); } });
    // ...
    return result;
}

```